### PR TITLE
feat: [sc-15955] Implement behavior.md in api documentation

### PR DESCRIPTION
**Pull request by AStoerkel1**
Story details: https://app.shortcut.com/clarkcan/story/15955

image

### CODE REVIEW

File: docs/competency-api/behavior.md [hunk 1]
```diff
@@ -4,4 +4,58 @@ title: Behavior
...
+}
+```
+
+### Request
+
+| Name | Required | Type | Description |
+|:----:|:-----:|:----:|-----|
+| task | true | string | The task to be carried out by the student |
+| details | false | string | Any details related to how the task should be completed |
+| work_role | false | string | Work role(s) in which this task could be performed |
```
Comment (pmanal1) [inline docs/competency-api/behavior.md]: I think we'll need to make a story to update the DTO to accept `work_role`.

File: docs/competency-api/behavior.md [hunk 2]
```diff
@@ -4,4 +4,58 @@ title: Behavior
...
+#### patch:
+```http
+https://api.competency-constructor.clark.center/competencies/:competencyId/behavior
+```
+
+### Headers
+
+```json
+{
+    Bearer: "Bearer Example.bearer.token"
```
Comment (Cwagne17) [inline docs/competency-api/behavior.md]: authorization header instead of Bearer

File: docs/competency-api/behavior.md [hunk 3]
```diff
@@ -4,4 +4,58 @@ title: Behavior
...
+| details | false | string | Any details related to how the task should be completed |
+| work_role | false | string | Work role(s) in which this task could be performed |
+| userId | true | Object Id: string | The userId matching the Bearer token |
+
+<details closed markdown="block">
+  <summary>
+    Example
+  </summary>
+```json
+{
```
Comment (Cwagne17) [inline docs/competency-api/behavior.md:43]: curl request example

**Review by Cwagne17 — APPROVED** [review]

**Review by Cwagne17 — CHANGES_REQUESTED** [review]
Previous requests. accidently approved.

File: docs/competency-api/behavior.md (hunk 1 above)
Comment (JosueMolinaMorales) [reply docs/competency-api/behavior.md]: Theses changes have been made in mike's recent PR.

Comment (JosueMolinaMorales) [inline docs/competency-api/behavior.md]: `work_role` is actually going to be an object id and only one can be sent i believe. Correct me if im wrong @mfranz1

**Review by JosueMolinaMorales — COMMENTED** [review]
Just a couple of things. Looks good though

**Review by pmanal1 — COMMENTED** [review]
Add a table for the query params for this doc and for the rest of them

**Comment by AStoerkel1** [comment]
> Add a table for the query params for this doc and for the rest of them

That seems a little unnececary this route does not have any query params, I'm making one for documentation which does tho

**Review by pzalep1 — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

File: docs/competency-api/behavior.md [hunk 4]
```diff
@@ -4,4 +4,69 @@ title: Behavior
...
+  -L "https://api.competency-constructor.clark.center/competencies/6112745b84804cf5833aa94c/behavior"
+```
+</details>
+
+### Response
+
+| Name | Description |
+|:----:|----|
+| 204 | Behavior updated successfully |
+| 400 | Competency is deprecated, rejected, or published and cannot be updated. Or body is incorrectly formatted. |
```
Comment (mfranz1) [inline docs/competency-api/behavior.md]: Can it be made more obvious that this is a list of 400 errors

**Review by mfranz1 — CHANGES_REQUESTED** [review]

**Review by Cwagne17 — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]

**Pull request merged**